Extension/src/LanguageServer/configurations.ts

```diff
@@ -1381,6 +1381,11 @@ export class CppProperties {
             result = result.replace(/\*/g, "");
         }
 
+        // Make sure all paths result to an absolute path
+        if (!path.isAbsolute(result)) {
```

Comment:
Some existing callers of resolvePath appear to expect that the result of calling resolvePath may be a relative path, so this may cause a regression in those cases (not sure). Can you add a bool parameter to enable the resolving a relative path to an absolute one so this can be enabled in the one call?

Reply:
You could also modify the pre-existing callers of resolvePath that appear to expect a relative path in some cases, but it seems like that might be more work.

Reply:
Or may not make this change in resolvePath and instead do it outside of it.

Reply:
@sean-mcmanus I think you're right that we have other callers that check if the path is relative and adjust it if so.  But I think that logic is not well thought-out and should all be consolidated, presumably into resolvePath.  We cannot rely on relative paths, because we cannot rely on the current working directory, as it may be changed by an API call, etc. (We can't expect it to point to be a particular workspace in a multiroot workspace).  I think we should remove all of the other handling of relative paths, in favor of resolving relative paths into absolutely paths early - as we process incoming user-specified path values.  That way, from that point on, we can assume we're working with absolute paths everywhere, and don't have to deal with them in multiple places.

Reply:
@Colengms Yeah, that sounds good to me (but more work, more code paths to verify don't regress, as mentioned previously).